core/src/test/java/org/apache/accumulo/core/clientImpl/lexicoder/ByteUtilsTest.java

```diff
@@ -70,4 +76,20 @@ public void testSplitWithOffset() {
     assertEquals(1, result.length);
     assertArrayEquals("5678".getBytes(), result[0]);
   }
+
+  @Test
+  public void testEscape() {
+    byte[] bytes = {0x00, 0x01};
+    byte[] escaped = ByteUtils.escape(bytes);
+    assertArrayEquals(bytes, ByteUtils.unescape(escaped));
+
+    // no escaped bytes found so returns the input
+    byte[] notEscaped = {0x02, 0x02, 0x02};
+    assertArrayEquals(notEscaped, ByteUtils.unescape(notEscaped));
+
+    // incomplete bytes would cause an ArrayIndexOutOfBounds in the past
+    exception.expect(IllegalArgumentException.class);
+    byte[] errorBytes = {0x01};
+    assertArrayEquals(errorBytes, ByteUtils.unescape(errorBytes));
```

Comment:
Is it expected that `ByteUtils.unescape(errorBytes)` will throw an exception?  If so, why check the return value?

Reply:
Yes, to avoid the ignored return warning.

Reply:
I guess could check that what is returned is not null?

Reply:
Whatever you think is best, I didn't realize this was being done to avoid a warning.  It confused me when I read the test because I was not sure what the expectation was.

Reply:
Ah OK.  I will the test more clear.  I can even make the exception its on unit test, which would probably help.

Reply:
You could do something like this instead, which is a little more clear:

```java
exception.expect(IllegalArgumentException.class);
byte[] escapedBytes = ByteUtils.unescape(errorBytes);
fail(String.valueOf(escapedBytes) + " should not have been assigned to escapedBytes");
```